src/main/java/picard/sam/markduplicates/ElcHashBasedDuplicatesFinder.java

```diff
@@ -58,92 +62,85 @@ void searchDuplicates(List<PairedReadSequence> sequences, Histogram<Integer> dup
         populateDupCandidates(sequences);
 
         //Duplicate set to filter the treated PairedReadSequences out
-        Set<PairedReadSequence> dupSet = new HashSet<>();
+        final Set<PairedReadSequence> dupSet = new HashSet<>();
+
         for (PairedReadSequence lhs : sequences) {
             if (dupSet.contains(lhs)) continue;
-
             final List<PairedReadSequence> dupes = new ArrayList<>();
-
-            for (PairedReadSequence rhs : lhs.dupCandidates) {
+            for (PairedReadSequence rhs : getSimilarReads(lhs)) {
                 if (dupSet.contains(rhs)) continue;
-
                 if (isDuplicate(lhs, rhs)) {
                     dupes.add(rhs);
                 }
             }
-
             dupSet.addAll(dupes);
             fillHistogram(duplicationHisto, opticalHisto, lhs, dupes);
         }
     }
 
+    private Set<PairedReadSequence> getSimilarReads(PairedReadSequence lhs) {
```

Comment:
why `lhs` for the name of the variable? please use something more fitting. also, `final`